transmart-core-db/src/integration-test/groovy/org/transmartproject/db/multidimquery/AggregateDataServicePgSpec.groovy

```diff
@@ -198,4 +203,123 @@ class AggregateDataServicePgSpec extends Specification {
         includingSecuredRecords['DEMO:POB'].nullValueCounts == 0
     }
 
+    void 'test caching counts per concept'() {
+        given: "countsPerConcept cache is empty"
+        String countsPerConceptCacheName = 'org.transmartproject.db.clinical.AggregateDataService.countsPerConcept'
+        cacheManager.getCache(countsPerConceptCacheName).clear()
+
+        def user = usersResource.getUserFromUsername('test-public-user-1')
+        def admin = usersResource.getUserFromUsername('admin')
+        Constraint studyConstraint = new StudyNameConstraint(studyId: "EHR")
+
+        when: "I call countsPerConcept for study EHR"
+        aggregateDataService.countsPerConcept(studyConstraint, user)
+        def countsPerConceptCache = cacheManager.getCache(countsPerConceptCacheName)
+
+        then: "countsPerConcept cache contains counts for study EHR and non-admin user"
+        countsPerConceptCache.getNativeCache().size() == 1
+        countsPerConceptCache.getAllKeys()[0] == [new ObjectMapper().writeValueAsString(["type": "study_name", "studyId": "EHR"]),
```

Comment:
I'd say we don't need to check exact match of the keys here. It's other tests concern. Changing how the key is calculated or serialized would break many places instead of one.
IMHO checking that cache got a new entry after the call is sufficient. Testing it for other inputs is not necessary either.

hm, seems like `getNativeCache()` returns an object, by taking `size()` on it, we rely on the particular setup. But I don't see a better way at the moment (I am not sure how to inject a mock cache manager easily).

Reply:
true, tests are updated now